packages/ui-kit/src/components/TimeSeries/utils.ts

```diff
@@ -45,13 +45,13 @@ export function formatLabels(options: FormatLabelsOptions): string[] | undefined
   return formatter ? formatter(labels || []) : labels
 }
 
-export function getDefaultGranularity(timeRange?: TimeRangeInput) {
+export function getDefaultGranularity(timeRange?: TimeRangeInput): TimeSeriesGranularity | null {
   const relative = timeRange?.relative
 
   if (!relative) {
     // TODO(mroberts): In a future release, we should calculate this for absolute ranges, too.
     //   Actually, all of this logic should move to the backend.
-    return TimeSeriesGranularity.Day
+    return null
```

Comment:
Why setting this to null? Granularity is still a required field right? Did you change this in the backend?

Reply:
You are right. I will add it back, but this code is not correct.

What we should do: make `granularity` optional in the backend and default it to something reasonable for the requested time range (it shouldn't matter if the time range is absolute or relative).